src/services/CertificateGenerationService.ts

```diff
@@ -191,9 +193,26 @@ class CertificateGenerationService {
       Signature: {
         ImageType: "png",
         ImageData: signature,
-      },
+      }
     };
-    const { testTypes, vehicleType, systemNumber } = testResult;
+
+    const { testTypes, vehicleType, systemNumber, testHistory } = testResult;
+
+    if (testHistory) {
+      for (const history of testHistory) {
+        for (const testType of history.testTypes) {
+          if (testType.testCode === testTypes.testCode) {
```

Comment:
```suggestion
          if (testType.testNumber === testTypes.testNumber) {
```
`testNumber` is the field to use to uniquely identify a test within test types. I don't know if in the scope of certificates test code is the field to use, since the test code is a field that can change. Whether that is the intended behaviour for certificate replacements, I don't know.

Reply:
We do want to detect if the code has changed. It's possible that a user could create a test with the wrong code and no certificate is generated. When that test is amended to the correct code we don't want the certificate to be a replacement.